### ISSUE TITLE
Neovide arbitrarily resizes itself when loosing focus when frame is none on Wayland

### ISSUE THREAD

**Issue by mystilleef** [neovide/neovide#1556]
**Describe the bug**

When starting `Neovide` with no frame (i.e --frame none) in a Wayland environment, such as GNOME Shell, it randomly and arbitrarily resizes its window when it looses focus. It also fails to remember window position and size on startup.

**To Reproduce**
1. Start `Neovide` with no frame in a `Wayland` environment, `neovide --notabs --multigrid --frame none`
2. Use your window manager to cascade or tile `Neovide` to preferred window size and position.
3. Switch focus from `Neovide` to another window.
4. `Neovide` will resize itself.
5. Restart `Neovide` and it will fail to remember its size and position.

**Expected behavior**
In a `Wayland` environment, a frameless `Neovide` window should not arbitrarily resize itself. It should also always remember its previous window size and position when launched.

**Video**

https://user-images.githubusercontent.com/273399/192120164-3404d8e8-4e7d-4eba-8595-f6cb26dea4cb.mp4

**Desktop (please complete the following information):**
 - OS: Fedora 36 Linux (GNOME Shell Wayland)
 - Neovide Version: Latest Nightly
 - Neovim Version: Latest Nightly
 
**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

https://drive.google.com/file/d/1wGZIoktcRsVP9zKxi23Pl6509LoFc3Tk/view?usp=sharing

**Additional context**
My `Neovide` settings follow.

```lua
-- gui font
vim.g.gui_font_default_size = 12
vim.g.gui_font_size = vim.g.gui_font_default_size
vim.g.gui_font_face = "Fira Code Retina"
vim.opt.guifont = string.format("%s:h%s:cDEFAULT", vim.g.gui_font_face, vim.g.gui_font_size)

-- neovide
local animation_speed = 0.250 -- animation speed in seconds
vim.g.neovide_refresh_rate = 100
vim.g.neovide_confirm_quit = false
vim.g.neovide_cursor_antialiasing = true
vim.g.neovide_remember_window_size = true
vim.g.neovide_cursor_vfx_mode = "railgun"
vim.g.neovide_hide_mouse_when_typing = true
vim.g.neovide_cursor_trail_size = animation_speed
vim.g.neovide_cursor_animation_length = animation_speed
vim.g.neovide_scroll_animation_length = animation_speed * 2
```

I start `Neovide` as follows:

`neovide --notabs --multigrid --wayland-app-id neovide --frame none`

**Comment by mystilleef** [comment]
I've also noticed that the `--geometry` flag doesn't work when `--frame none` in `Wayland`. The geometry size is either ignored or a crash happens.

**Comment by fredizzimo** [comment]
I'm pretty sure that this is fixed in our latest 0.12.1 release
